.github/workflows/super-linter.yaml

```diff
@@ -21,6 +21,19 @@ jobs:
         with:
           # Full git history is needed to get a proper list of changed files within super-linter
           fetch-depth: 0
+        if: ${{ !contains(github.event_name, 'pull_request') }}
+      - name: Checkout PR Code
+        uses: actions/checkout@v3.3.0
+        with:
+          # Full git history is needed to get a proper list of changed files within super-linter
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Do you have any examples of the runner log saying that it is in a failed state because of the detached head? I was not able to find any in a brief search through my own fork and the upstream repo.

Reply:
If I'm reading the logs correctly, it doesn't fail because of the detached head, it PASSES because of the detached head.  I believe the linter is running on HEAD and if there is no HEAD (or it's detached) it runs on 0 files, finds 0 errors, and logs that as passing.

Reply:
Answering my own question; I think all we need is to set the ref on checkout. That seemed to work for me in my local repo.

```
      - name: Checkout Code
        uses: actions/checkout@v3.3.0
        with:
          # Full git history is needed to get a proper list of changed files within super-linter
          fetch-depth: 0
        if: github.event_name != 'pull_request'

      - name: Checkout PR Code
        uses: actions/checkout@v3.3.0
        with:
          # Full git history is needed to get a proper list of changed files within super-linter
          fetch-depth: 0
          ref: refs/heads/${{ github.head_ref }}
        if: github.event_name == 'pull_request'
```

I've also seen usages of `github.event.pull_request.head.sha` for the ref instead. Not sure which is better.

Reply:
@jpdahlke That works for me as well.  I took that idea combined with using a fallback value to reduce the number of steps and conditionals, and I think we've got a nice, elegant solution here :smiley: